gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/iceberg/IcebergDatasetFinder.java

```diff
@@ -74,18 +84,15 @@ public List<IcebergDataset> findDatasets() throws IOException {
     String dbName = properties.getProperty(ICEBERG_DB_NAME);
     String tblName = properties.getProperty(ICEBERG_TABLE_NAME);
 
-    try {
-      IcebergCatalog icebergCatalog = createIcebergCatalog(this.properties);
-      /* Each Iceberg dataset maps to an Iceberg table
-       * TODO: The user provided database and table names needs to be pre-checked and verified against the existence of a valid Iceberg table
-       */
-      matchingDatasets.add(createIcebergDataset(dbName, tblName, icebergCatalog, properties, sourceFs));
-      log.info("Found {} matching datasets: {} for the database name: {} and table name: {}", matchingDatasets.size(),
-          matchingDatasets, dbName, tblName);
-      return matchingDatasets;
-    } catch (ReflectiveOperationException exception) {
-      throw new IOException(exception);
-    }
+    IcebergCatalog sourceIcebergCatalog = createIcebergCatalog(this.properties, CatalogLocation.SOURCE);
+    IcebergCatalog destinationIcebergCatalog = createIcebergCatalog(this.properties, CatalogLocation.DESTINATION);
+    /* Each Iceberg dataset maps to an Iceberg table
+     * TODO: The user provided database and table names needs to be pre-checked and verified against the existence of a valid Iceberg table
+     */
+    matchingDatasets.add(createIcebergDataset(dbName, tblName, sourceIcebergCatalog, destinationIcebergCatalog, properties, sourceFs));
+    log.info("Found {} matching datasets: {} for the database name: {} and table name: {}", matchingDatasets.size(),
```

Comment:
Initially I remember we had added it because we plan to find multiple datasets and the return type also being a List. The current implementation however only caters to one matching dataset based on the `dbname` and `tableName`. In light to support a list of matching datasets taking in may be a `list of (dname, tableName)`... we added the size. Either way, we can remove it for now and add it later when we support multiple `IcebergDataset`. I have kept it as is in my latest commit though... but can get rid of it!

Reply:
ok, sounds reasonable to keep, if we have ambitions to support a list.

NBD, but I would probably add a clarifying comment along the lines,
> // until future support added to specify multiple icebergs, count expected always to be one